pom.xml

```diff
@@ -238,6 +243,8 @@
           <target>1.8</target>
           <compilerArgs combine.children="override">
             <compilerArg>-Xep:FutureReturnValueIgnored:OFF</compilerArg>
+            <!-- Too limited, does not detect locks acquired by a calling method: -->
+            <compilerArg>-Xep:GuardedBy:OFF</compilerArg>
```

Comment:
For example, `RateLimitingRequestThrottler` would cause errors in `acquire` (even though we assert that the lock is held at the beginning of the method).

Reply:
Can't we instead reacquire the lock in these cases? According to the javadocs of `ReentrantLock`, if the thread already has the lock, acquiring it again is a no-op, and `isHeldByCurrentThread()` should be used for debugging purposes only.

Reply:
Reacquiring the lock is not the same as expecting it to be held. The javadocs explicitly mention assertions in callees as a typical use case.